wikimedia/wikipedia-ios
# Update search's language bar when languages change

Fixes phabricator ticket: https://phabricator.wikimedia.org/T252912

This updates the language bar in the search screen whenever the languages change. (`if let selectedLang = currentlySelectedSearchLanguage` shouldn't return nil in any expected situation, I believe, which is why I didn't worry about that situation.)

We're spamming a refresh every time a language changes, when in some cases - if the user has only added and/or deleted languages beyond the first three on the table - it isn't needed. Given the rarity of that situation combined with moderate complexity of that check*, this seemed like an appropriate way to handle it. If someone feels we should do the checks*, let me know and I'm happy to update this.

*Checks would be: confirm that for the languages in `languageBarLanguages()`, their `localizedName`s are the button's titles (in order). Only need to check up to first 3 languages, however. Also, need to check that if there are fewer than 3 languages, the number of buttons is identical to the number of languages (in case the user deleted a language).

**Test steps**
1. Start with 2 wikipedia languages active in the app.
1. Open any search screen.
1. Tap `More` button, and tap `Add another language` to add a third language.
1. Tap the `X` button to return to the search screen.
1. Confirm the third language is shown under the search box.
1. Tap `More` button, and delete that third language from the list.
1. Tap `Add another language` to add a (different) third language.
1. Tap the `X` button to return to the search screen.
1. Confirm the new third language is shown under the search box.
1. Tap `More` button, and delete that third language from the list.
1. Tap the `X` button to return to the search screen.
1. Confirm that only two languages are shown under the search box.
